artifact/metadata.go

```diff
@@ -287,9 +288,105 @@ func (ti *TypeInfo) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
-type TypeInfoDepends map[string]string
+type TypeInfoDepends map[string]interface{}
 
-type TypeInfoProvides map[string]string
+func (t TypeInfoDepends) Map() map[string]interface{} {
+	return map[string]interface{}(t)
```

Comment:
What's the motivation behind this function? You're not using it anywhere, and this appears to return a copy. Isn't it better to let the user cast and decide whether to copy or not?

Same with the other `Map()` function.

Reply:
It is used in the code for `MEN-2549`. Perhaps I should add it in there instead ;)

Reply:
If you know where you will use it, then I'm fine with it.